src/NServiceBus.Core/Reliability/Outbox/OutboxDeduplicationBehavior.cs

```diff
@@ -70,7 +70,7 @@ void DispatchOperationToTransport(IEnumerable<TransportOperation> operations, Co
                 var deliveryConstraints = deliveryConstraintsFactory.DeserializeConstraints(transportOperation.Options)
                     .ToList();
             
-                dispatchStrategy.Dispatch(dispatcher, message, routingStrategy, new AtLeastOnce(), deliveryConstraints, context);
+                dispatchStrategy.Dispatch(dispatcher, message, routingStrategy, new AtLeastOnce(), deliveryConstraints, context).GetAwaiter().GetResult();
```

Comment:
Did you see the `GetAwaiter().GetResult()`? It will be when the pipeline is async